### ISSUE TITLE
Add "This is a corporate account" indicator

### ISSUE THREAD

**Issue by ocdtrekkie** [mastodon/mastodon#20804]
### Pitch

Add a profile setting to indicate the account is a corporate account or account run by a corporate entity. (It's plausible this may be a good indicator for "named" accounts posting in an official capacity, but I am unsure on this.)

### Motivation

A lot of people are particularly concerned about corporations/brands making accounts on the fediverse. It's definitely true that there is good and bad behavior in this space, and I think it's worth supporting corporate accounts because the alternative is the continuation of proprietary/closed social networks.

However, I think we should ask them to, like bots, indicate they are corporate-run accounts. This would allow users to filter them if they felt so inclined without requiring them to be on their own server (Hachyderm notably has accepted a corporate account now, for instance) or requiring one to defederate entire servers to block or filter corporate accounts. While servers may have their own policies about what level of "advertising" is acceptable, a corporate flag would present a network-wide default minimum: Ads should be from accounts marked corporate.

**Comment by ocdtrekkie** [comment]
@downeymj I think it'd both be crucial that these are self-selected by the account creator, and that they are able to be universally recognized by other servers across the ecosystem, similar to Bots. Client apps will likely want to explicitly build settings and behavior around this designation, and for those settings to apply to content they see from servers across the entire federation.

**Comment by ocdtrekkie** [comment]
Another note, actually, as I explore: https://superuser.com/questions/1439448/what-is-the-effect-of-ticking-this-is-a-bot-account-in-mastodon

ActivityPub explicitly supports an actor type field, and normal Mastodon accounts use "Person", and bots use "Service". This designation would properly switch the actor type to "Organization". (It's possible this should setting should use the "organization" language rather than "corporate" as well, to avoid ambiguity on if nonprofits or open source projects should or should not use this setting.)

**Comment by CEbbinghaus** [comment]
> It's possible this should setting should use the "organization" language rather than "corporate" as well, to avoid ambiguity on if nonprofits or open source projects should or should not use this setting.

There is a big difference between a Open source project promoting its new release and a Company trying to sell you its product. NonProfits seem to occupy the exact middle ground between those worlds but overall there is a large gap between what kinds of Organizations exist. If I was to filter out all Organizations because I don't want to see a McDonalds ad that doesn't mean I don't want to see my favorite framework release its much anticipated 1.0.0 version.

Overall I think we are at a very crucial point of adoption where our choices of how we accept brands and "advertising" in all its forms will have a meaningful impact far down the road. And while I agree that the `organization` umbrella is probably a good distinguishment it is a bit broad and lumps small media outlets of 2-3 people together with megacorporation's.

**Comment by ocdtrekkie** [comment]
True but that isn't on the platform or protocol to mediate: Mastodon should simply provide the distinction between Persons and Organizations (Groups also exists in ActivityPub, which may also apply to smaller projects?) and it's up to the community to decide how to interact with them.

Presumably also, you wouldn't "block all organizations", you might just filter them out of your feed. Which wouldn't stop you from following ones you were interested in hearing from.

**Comment by n0toose** [comment]
> This designation would properly switch the actor type to "Organization".

I ran an account for an open-source project with ~1k followers and believe this should be useful and protocol-respecting.

**Comment by Andonome** [comment]
I don't see much point.

You can tell who's legit by the website verification.  And you can see who's peddaling items by the pedal.

There's also no clear line of demarkation - a lot of people on hobby instances are talking about their kickstarter or sales, but they're not there as a brand - they're there as a person.

Given that we've not seen any actual companies on Mastodon, it seems better to wait rather than preemptiviely add solutions for a problem that doesn't exist.

**Comment by CEbbinghaus** [comment]
> Presumably also, you wouldn't "block all organizations", you might just filter them out of your feed. Which wouldn't stop you from following ones you were interested in hearing from.

I meant filtering out organizations (a filter is pretty much blocking/CW'ing any content). My point is that the broad definition of organization makes for a pretty bad filter. 

Overall I think this distinction is important and will provide a framework we can start working within to clearly distinguish between users and companies. As for how we want companies interacting with this platform is a completely different matter and a little outside the scope of this Issue

**Comment by ocdtrekkie** [comment]
@Andonome There are multiple corporate accounts joining the fediverse on a regular basis. Tailscale started participating today, for example.

This is NOT about verification. This is for automatic classification. It's to allow clients and servers the ability to automatically distinguish organizations from persons.

**Comment by Danie10** [comment]
We can certainly expect in future that all sorts of organisations will make a home on Mastodon, such as commercial organisations, non-profits, cultural, governmental, emergency rescue, and many more.

I'm not sure though how automatic classification may happen?

We'd also need to think about the reason for doing the classification, and just be sure whether it will have the intended outcomes. For example, bots can be good or bad eg. a bot replicating the BBC news is something I'd like to follow if the BBC is not posting themselves. However, bots selling crypto I don't want to see. So do I filter all bots, or mute them/their instance as they appear, etc.

A difference on Mastodon is that unlike Twitter, organisations (or anyone) should not be able to use money to push something into my feed. So if company XYZ establishes themselves and starts posting, would I see anything from them if I don't follow them? If I do see something boosted, can I just mute them and they "disappear"?

**Comment by CEbbinghaus** [comment]
There is also the argument to be made that organizations should be setting up their own instances. It is important to have a distinction for organization accounts (e.g a project account on a foss instance) however most organizations of any scale should be running under their own domain. This is what the Texas Observer did: https://texasobserver.social/public/local and its something that should be more widely adopted. There is no way to currently class an instance as a "organization" instance so together with the change to mark users as organizations it might be worth looking at adding a instance classifier too

**Comment by tdelmas** [comment]
> There is also the argument to be made that organizations should be setting up their own instances. 
@CEbbinghaus it might be true for large organizations that either expect a lot of their members to join (e.g. universities, large newspapers, ...), or that have the sufficient funds to maintain it for 1-10 accounts (so, large commercial entities).

For smalls non-profit, it doesn't make sense to set up an instance for only one account representing the organization (or only to host 2-3 accounts for their members).

**Comment by ocdtrekkie** [comment]
Yeah, I think there's definitely good reasons for both approaches, but as long as all of the above marks their accounts as organization accounts, ActivityPub clients wouldn't need special logic to detect both organization servers and organization accounts on other servers: The design to mark a given account as an organization is already in the spec. (Presumably the simplest way to mark a private corporation's server as such would just be a setting that automatically marks all it's accounts as organization accounts.)

**Comment by trwnh** [comment]
Would it make more sense to refactor the bot flag into an "account type" selector (so you can choose between Person / Service / Organization)? Or keep it as flags and publish in a type array (so you can be both Service + Organization)?

**Comment by ocdtrekkie** [comment]
I don't believe ActivityPub would support a way of reflecting both simultaneously, so a type selector might make more sense. But I'm not qualified to have a strong position on implementation details.

**Comment by trwnh** [comment]
ActivityPub supports it just fine -- it's implementations that may have written fragile code because they didn't account for this valid syntax. Needs a summary of how most projects have gone about it so far.

**Comment by VyrCossont** [comment]
This would be a great addition to #24055 account search.

**Comment by ghost** [comment]
I just wanted to add that I think this is needed more now than before. I've recently been seeing lots of commercial accounts promoting their products lately and it's been a struggle to block them all. It's often companies creating posts with a dozen hashtags and a linking to products. It's not always the rule-breaking spam either, sometimes it's just regular marketing copy/links, so I can't always report it as spam. I've noticed it happening primarily on [mastodon.social's live feed](https://mastodon.social/public/local) (probably due to its size).

The joinmastodon site says "No ads to waste your time" which is not true at all anymore. These types of posts are ads. 

And the site also says "Your home feed should be filled with what matters to you most, not what a corporation thinks you should see." But these corporate marketing teams are really thinking that I should see what their latest offerings are. 😐
